src/main/java/seedu/address/model/task/TaskDeadlineContainsDatePredicate.java

```diff
@@ -0,0 +1,27 @@
+package seedu.address.model.task;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Task}'s {@code TaskDeadline} matches the date given.
+ */
+public class TaskDeadlineContainsDatePredicate implements Predicate<Task> {
+    private final TaskDeadline date;
+
+    public TaskDeadlineContainsDatePredicate(TaskDeadline date) {
+        this.date = date;
+    }
+
+    @Override
+    public boolean test(Task task) {
+        return date.equals(task.getDeadline());
```

Comment:
can filter the task by dateline before instead of equal? We want to get all the tasks before a particular dateline.

Reply:
Resolved in the "Update filter by deadline function" commit!